[2025-07-01T12:13:48.559+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-07-01T12:13:48.573+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job manual__2025-07-01T12:13:47.645309+00:00 [queued]>
[2025-07-01T12:13:48.576+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job manual__2025-07-01T12:13:47.645309+00:00 [queued]>
[2025-07-01T12:13:48.576+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2025-07-01T12:13:48.580+0000] {taskinstance.py:2330} INFO - Executing <Task(SparkSubmitOperator): submit_grazing_detector_job> on 2025-07-01 12:13:47.645309+00:00
[2025-07-01T12:13:48.584+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'grazing_detection_streaming_cluster', 'submit_grazing_detector_job', 'manual__2025-07-01T12:13:47.645309+00:00', '--job-id', '7', '--raw', '--subdir', 'DAGS_FOLDER/grazing_dag.py', '--cfg-path', '/tmp/tmpniiwj9_p']
[2025-07-01T12:13:48.585+0000] {standard_task_runner.py:91} INFO - Job 7: Subtask submit_grazing_detector_job
[2025-07-01T12:13:48.584+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=260) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-01T12:13:48.585+0000] {standard_task_runner.py:63} INFO - Started process 261 to run task
[2025-07-01T12:13:48.605+0000] {task_command.py:426} INFO - Running <TaskInstance: grazing_detection_streaming_cluster.submit_grazing_detector_job manual__2025-07-01T12:13:47.645309+00:00 [running]> on host 3ceedc259e60
[2025-07-01T12:13:48.634+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='grazing_detection_streaming_cluster' AIRFLOW_CTX_TASK_ID='submit_grazing_detector_job' AIRFLOW_CTX_EXECUTION_DATE='2025-07-01T12:13:47.645309+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-01T12:13:47.645309+00:00'
[2025-07-01T12:13:48.634+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-07-01T12:13:48.644+0000] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2025-07-01T12:13:48.644+0000] {spark_submit.py:340} INFO - Spark-Submit cmd: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092
[2025-07-01T12:13:48.678+0000] {spark_submit.py:491} INFO - JAVA_HOME is not set
[2025-07-01T12:13:48.678+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-07-01T12:13:48.683+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 156, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092. Error code is: 1.
[2025-07-01T12:13:48.685+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=grazing_detection_streaming_cluster, task_id=submit_grazing_detector_job, run_id=manual__2025-07-01T12:13:47.645309+00:00, execution_date=20250701T121347, start_date=20250701T121348, end_date=20250701T121348
[2025-07-01T12:13:48.688+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 7 for task submit_grazing_detector_job (Cannot execute: spark-submit --master spark://spark-master:7077 --packages org.apache.spark:spark-sql-kafka-0-10_2.13:3.5.1 --name arrow-spark /opt/***/spark_jobs/grazing_detector.py kafka:29092. Error code is: 1.; 261)
[2025-07-01T12:13:48.729+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-07-01T12:13:48.738+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-01T12:13:48.739+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
